exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/__init__.py

```diff
@@ -141,18 +141,21 @@ def collect(self) -> None:
         if self._callback is not None:
             self._callback()
 
+        metric_family_id_metric_family = {}
+
         while self._metrics_to_export:
             for export_record in self._metrics_to_export.popleft():
-                prometheus_metric = self._translate_to_prometheus(
-                    export_record
+                self._translate_to_prometheus(
+                    export_record, metric_family_id_metric_family
                 )
-                if prometheus_metric is not None:
-                    yield prometheus_metric
+
+            if metric_family_id_metric_family:
+                for metric_family in metric_family_id_metric_family.values():
+                    yield metric_family
 
     def _translate_to_prometheus(
-        self, metric: Metric
-    ) -> Optional[core.Metric]:
-        prometheus_metric = None
+        self, metric: Metric, metric_family_id_metric_family
```

Comment:
can you add a type for metric_family_id_metric_family

Reply:
Added